src/cpr_data_access/models/search.py

```diff
@@ -43,6 +46,23 @@ def query_string_must_not_be_empty(cls, query_string):
             raise QueryError("query_string must not be empty")
         return query_string
 
+    @field_validator("family_ids", "document_ids")
+    def ids_must_fit_pattern(cls, ids):
+        """
+        Validate that the family and document ids are ids.
+
+        Example ids:
+            CCLW.document.i00000004.n0000
+            CCLW.family.i00000003.n0000
+            CCLW.executive.10014.4470
+            CCLW.family.10014.0
+        """
+        if ids:
+            for i in ids:
```

Comment:
https://github.com/climatepolicyradar/navigator-pipeline-report/blob/main/src/cpr_pipeline_reporter/config.py

Line 47ish there are some patterns you might find useful!

Reply:
as pipeline reporter depends on DAL it might make sense to centralise this pattern in DAL in a future piece of work?

Reply:
Thanks both! This is a good shout, I'll have a look at integrating this now and think through a longer term location for it as a separate change

Reply:
This is now done